### PR TITLE
Add deepseek models

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 1]
```diff
@@ -2355,6 +2363,7 @@ async def _infer_hardware(
     config_map = await _get_recommended_hardware_config_map()
     by_model_name = {item["name"]: item for item in yaml.safe_load(config_map["byModelName"])}
     by_gpu_memory_gb = yaml.safe_load(config_map["byGpuMemoryGb"])
+    print(min_memory_gb)
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:2366]: logger.info() or remove?

**Review by seanshi-scale — APPROVED** [review]
one small nit with a print() but lgtm after that's handled

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py (hunk 1 above)
Comment (dmchoiboi) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:2366]: oh woops; will remove

File: charts/model-engine/values_circleci.yaml [hunk 2]
```diff
@@ -270,12 +270,36 @@ recommendedHardware:
...
       gpus: 2
       memory: 40Gi
       storage: 40Gi
+      gpu_type: nvidia-hopper-h100
+    - name: deepseek-coder-v2-lite
+      cpus: 160
+      gpus: 8
+      memory: 800Gi
+      storage: 640Gi
+      gpu_type: nvidia-hopper-h100
```
Comment (yunfeng-scale) [inline charts/model-engine/values_circleci.yaml:287]: i think this is optional and we don't need them here. if we intend to add them, `deepseek-coder-v2-lite` doesn't need so much resource

File: model-engine/tests/unit/domain/test_llm_use_cases.py [hunk 3]
```diff
@@ -1937,6 +1937,130 @@ async def test_validate_checkpoint_files_safetensors_with_other_files():
...
+    hardware = await _infer_hardware(
+        fake_llm_artifact_gateway, "deepseek-coder-v2-lite-instruct", ""
+    )
+    assert hardware.cpus == 160
+    assert hardware.gpus == 8
+    assert hardware.memory == "800Gi"
+    assert hardware.storage == "640Gi"
+    assert hardware.gpu_type == GpuType.NVIDIA_HOPPER_H100
+
+    hardware = await _infer_hardware(
```
Comment (yunfeng-scale) [inline model-engine/tests/unit/domain/test_llm_use_cases.py:2055]: hmm i'm surprised this test passed

File: charts/model-engine/values_circleci.yaml (hunk 2 above)
Comment (dmchoiboi) [reply charts/model-engine/values_circleci.yaml:287]: oh, i was just looking at the instructions on their repo: https://huggingface.co/deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct#5-how-to-run-locally

Comment (yunfeng-scale) [reply charts/model-engine/values_circleci.yaml:287]: they just said "If you want to utilize DeepSeek-Coder-V2 in BF16 format for inference, 80GB*8 GPUs are required" right? i think 8x80GB only applies to DeepSeek-Coder-V2 not lite

**Pull request merged**